### PR TITLE
DEX: Add swap functionality

### PR DESCRIPTION
**Pull request by 0xalecks**

### CODE REVIEW

**Review by marshall-bean — APPROVED** [review]
Minor comments that can be addressed in the next PR.

File: .yarn/versions/485022b5.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+undecided:
```
[the remark this answers is not in the document]
Comment (0xalecks) [reply .yarn/versions/485022b5.yml:1]: from a `yarn version patch` in the sdk-core project

File: projects/sdk-core/src/lib/TokenValue.ts [hunk 2]
```diff
@@ -140,10 +141,18 @@ export class TokenValue {
...
-  public toHuman(): string {
-    return this.value.toString();
+  public toHuman(format?: string): string {
+    if (!format) return this.value.toString();
+
+    if (format !== "auto") {
+      return numeral(this.value.toString()).format(format).toUpperCase();
+    }
+
+    return "";
```
[the remark this answers is not in the document]
Comment (0xalecks) [reply projects/sdk-core/src/lib/TokenValue.ts:155]: fixed

File: projects/sdk-wells/src/lib/swap/Quote.ts [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+    const pipelineAddress = "0xb1be0000bfdcddc92a8290202830c4ef689dceaa";
+    const steps = this.steps;
+
+    const desiredAmount = this.amountUsedForQuote; // Amount desired
+    const operations = [];
+
+    for (let i = 0; i < steps.length; i++) {
+      const step = this.steps[i];
+      const nextStep = this.steps[i + 1] || null;
+      // console.log("Step:", step.fromToken.symbol, " -> ", step.toToken.symbol);
```
[the remark this answers is not in the document]
Comment (0xalecks) [reply projects/sdk-wells/src/lib/swap/Quote.ts:205]: leaving these in for now, need to add some debugging later and this will remind me where and what

**Pull request merged**